src/circuit/layouter.rs

```diff
@@ -152,6 +171,17 @@ impl<F: Field> RegionLayouter<F> for RegionShape {
         })
     }
 
+    fn assign_advice_from_constant<'v>(
+        &'v mut self,
+        annotation: &'v (dyn Fn() -> String + 'v),
+        column: Column<Advice>,
+        offset: usize,
+        constant: Assigned<F>,
+    ) -> Result<Cell, Error> {
+        // The rest is identical to witnessing an advice cell.
+        self.assign_advice(annotation, column, offset, &mut || Ok(constant))
```

Comment:
Does this mean that the `assign_advice_from_instance` implementation just below this could be simplified as well?

Reply:
There was a reason I decided not to do that at the time, but I don't remember what it was. I might be thinking of a different `assign_advice_from_instance` impl though. AFAICT `RegionShape::assign_advice_from_instance` could indeed be simplified like this.

Reply:
Ah, this might have been it:
```
error[E0283]: type annotations needed
   --> src/circuit/layouter.rs:196:14
    |
196 |         self.assign_advice(annotation, advice, offset, &mut || unreachable!())
    |              ^^^^^^^^^^^^^ cannot infer type for type parameter `F` declared on the trait `RegionLayouter`
    |
    = note: cannot satisfy `_: Field`
```